File: src/backend/Data/Entities/Agent.cs
namespace CocktailMaker.Data.Entities;

public class Agent
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string AgentId { get; set; } = string.Empty;
    public bool IsOnline { get; set; }
    public DateTime? LastSeen { get; set; }
}

File: src/frontend/src/contexts/Agent.ts
export interface IAgent {
  id: number
  name: string
  agentId: string
  isOnline: boolean
  lastSeen: string | null
}

export interface IAgentPump {
  pumpIndex: number
  ingredientId: number | null
  ingredientName: string | null
}

export interface IUpdatePumpSlot {
  pumpIndex: number
  ingredientId: number | null
}

export type AgentContextType = {
  agents: IAgent[]
  agentPumps: Record<number, IAgentPump[]>
  dispense: (recipeId: number) => Promise<void>
  fetchAgentPumps: (agentId: number) => Promise<void>
  updateAgentName: (id: number, name: string) => Promise<void>
  updateAgentPumps: (id: number, pumps: IUpdatePumpSlot[]) => Promise<void>
}
